### PR TITLE
 Updated the meshkit error for performance_profiles_handler and provider_handler

### LINKED ISSUE #3304: [Chore] Update performance profile, provider, and schedule handlers to use new Meshkit Errors
Note: This is a child issue of #3275 
#### Current Behavior
Meshery Server (and its other components, like Adapters) use a centralized library, MeshKit, to provide uniform utilities (e.g. raising and logging errors). Currently, Meshery Server uses a now antiquated style of error logging with logrus.

#### Desired Behavior
Ideally, all use of logrus is replaced with use of MeshKit-style error logging.

#### Implementation
Meshery Handlers needs to be:
 - Updated to use meshkit errors. 
 - Switch to handler Log instead of `logrus`

Handlers to update:

- [ ] performance profiles handler
- [ ]  provider handler
- [x]  schedule handler

For reference, you can checkout  the [`grafana handler`](https://github.com/layer5io/meshery/blob/master/handlers/grafana_handlers.go) and [`prometheus handler`](https://github.com/layer5io/meshery/blob/master/handlers/prometheus_handlers.go) or the [docs](https://docs.meshery.io/project/contributing-error).
Define all Error functions for all the handlers to be written in the `handlers/error.go` file.

#### Mockups

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by Shreyas220**
Signed-off-by: Shreyas220 <shreyas.ny@gmail.com>

**Description**
 Updated the meshkit error for performance_profiles_handler and provider_handler

This PR fixes #3304

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Shreyas220** [comment]
@sayantan1413

File: handlers/performance_profiles_handler.go [hunk 1]
```diff
@@ -71,7 +78,9 @@ func (h *Handler) GetPerformanceProfilesHandler(
 
 	resp, err := provider.GetPerformanceProfiles(r, q.Get("page"), q.Get("page_size"), q.Get("search"), q.Get("order"))
 	if err != nil {
-		http.Error(rw, fmt.Sprintf("failed to fetch the performance profiles: %s", err), http.StatusInternalServerError)
+		obj := "performance profile"
+		//get query performance profile
```
Comment (sayantan1413) [inline handlers/performance_profiles_handler.go:82]: Hey @Shreyas220 you can log the error here.

**Review by sayantan1413 — CHANGES_REQUESTED** [review]
Hey @Shreyas220 consider updating the `component_info.json` file by updaing the `next_error_code` to `2048`.

**Comment by Shreyas220** [comment]
> Hey @Shreyas220 consider updating the `component_info.json` file by updaing the `next_error_code` to `2048`.  Rest it looks good to me. 😊

 @sayantan1413 ,Why shift it to 2048 ¿
I see no conflicting error

**Comment by sayantan1413** [comment]
Hey @Shreyas220 according to your latest change in the `handler/error.go` the last int code used here is "2047" and as you have manually numbered the error codes so you need to update the `next_error_code` to "2048" . Now whenever there will be new error code it will get automatically replaced by "2048" when you run the command `make error`. Actually the `next_error_code` denotes the latest int code that should be used as the next error code.

File: handlers/error.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ const (
 	ErrRetrieveUserTokenCode    = "2044"
 	ErrFailToSaveCode           = "2045"
 	ErrFailToDeleteCode         = "2046"
-	ErrFailToLoadExtensionsCode = "2047"
+	ErrFailToLoadExtensionsCode = "2048"
```
Comment (sayantan1413) [inline handlers/error.go:55]: Hey @Shreyas220 let `ErrFailToLoadExtensionsCode = "2047"` only change the `https://github.com/meshery/meshery/blob/master/component_info.json ` file `next_error_code` to "2048".

**Review by sayantan1413 — CHANGES_REQUESTED** [review]

File: handlers/error.go (hunk 2 above)
Comment (Shreyas220) [reply handlers/error.go:55]: oh, so sorry ......didn't read the comment properly  :p

**Review by sayantan1413 — DISMISSED** [review]
LGTM. @kumarabd have a review on this.

**Comment by leecalcote** [comment]
@DelusionalOptimist I wonder how we might avoid merge conflicts between PRs for error codes. // @mgfeller

**Comment by mgfeller** [comment]
> @DelusionalOptimist I wonder how we might avoid merge conflicts between PRs for error codes. // @mgfeller

Can't visualize the merge conflict you're referring to? 

Not sure whether it is related, but the sequence of entries in lists (like `DuplicateNames`) in the summary output will be sorted in the next version of errorutil.

**Comment by leecalcote** [comment]
Oh, @mgfeller, it seems @Shreyas220 is quickly (very good) eliminated the merge conflict. There’s another open PR with error analysis conflict. I’ll find it…

**Comment by sayantan1413** [comment]
Hey @mgfeller like suppose if two of us are working together and making the changes in the error.go file. Then like if I mentioned a error code `ErrMe` and after running make error I get the int code of `ErrMe` as "1000" in line `number 1` and again the other person working in the same file defined a error like `ErrProgram` and he also got the errror code as "1000" . in `line number 1 `After making the changes both of us pushed the code. If suppose my code merges first then the other contributor will get a huge set of merge conflict in the three JSON files as well as the `component_info.json` and the error.go file.

**Comment by mgfeller** [comment]
Hey @sayantan1413 yes, absolutely, makes sense. The json files could just be recreated by running errorutil (analyze / update). Conflicting / duplicate codes are probably new ones (i.e. not yet published), and one of them could be replaced by "replace_me", then updated using the errorutil. `next_error_code` might need to be handled manually, the next version of errorutil will issue an error if it is lower than the highest code.

My impression is that it is not often that several contributors are working in the same package though, but I have no statistics here :-) 

We could observe a bit more and discuss possible measures in a while?

**Comment by Shreyas220** [comment]
A temporary solution would be creating a common docs for errors, where everyone, before making a PR request, would update the error code and name so we could check if a particular error code is being used ?

**Comment by mgfeller** [comment]
> A temporary solution would be creating a common docs for errors, where everyone, before making a PR request, would update the error code and name so we could check if a particular error code is being used ?

Hm yes, however, one of the objectives was to avoid centralized handling of error codes, automating everything. Also, this might stick and be difficult to get rid of. I think incorporating in the CI as discussed [here on Slack](https://layer5io.slack.com/archives/C011VPREG2Z/p1625812409203500) is best, and educate us contributors to not set error codes but use strings.

**Comment by leecalcote** [comment]
@Shreyas220 @mgfeller @sayantan1413, the "how to use the MeshKit Errors function" is currently a light write-up, and would benefit from some of your additional perspectives, I'm sure - https://docs.meshery.io/project/contributing-error. Please indulge by expanding on any particulars.

Mind you, "How the MeshKit Error Code Utility works" is a separately documented topic and not intended for the "contributing-error" page, per say. We keep that doc separate because the vast majority of contributors need to know how to use the library, not how it works. That separate doc expects the reader to have at least a Masters-level comprehension of programming languages or a long, gray beard.

**Comment by Shreyas220** [comment]
> @Shreyas220 @mgfeller @sayantan1413, the "how to use the MeshKit Errors function" is currently a light write-up, and would benefit from some of your additional perspectives, I'm sure - https://docs.meshery.io/project/contributing-error. Please indulge by expanding on any particulars.
> 
> Mind you, "How the MeshKit Error Code Utility works" is a separately documented topic and not intended for the "contributing-error" page, per say. We keep that doc separate because the vast majority of contributors need to know how to use the library, not how it works. That separate doc expects the reader to have at least a Masters-level comprehension of programming languages or a long, gray beard.

yes @leecalcote ofcourse 
i found an issue related to updating the eroror docs(3029), ill continue there 😄

**Comment by Shreyas220** [comment]
also @sayantan1413  @kumarabd, I think we can merge this branch ??😅

**Review by sayantan1413 — APPROVED** [review]
Yes @Shreyas220 it looks good to me. @kumarabd please have a review on this.

**Review by kumarabd — APPROVED** [review]
LGTM

**Pull request merged**